mathics/builtin/lists.py

```diff
@@ -2120,4 +2120,99 @@ class Total(Builtin):
     rules = {
         'Total[head_]': 'Apply[Plus, head]',
         'Total[head_, n_]': 'Apply[Plus, Flatten[head, n]]'
-    }
\ No newline at end of file
+    }
+
+
+class Mean(Builtin):
+    """
+    <dl>
+    <dt>'Mean[$list$]'
+    <dd>
+        Mean over all values in list (Total[list] / Length[list])
+    </dd>
+    </dt>
+    </dl>
+    >> Mean[{1,2,3}]
+     = 2
+    >> Mean[{a, b, 213}]
+     = (123 + a + b) / 3
+    """
+    rules = {
+        'Mean[list_List]': 'Total[list] / Length[list]'
+    }
+
+
+class Variance(Builtin):
+    """
+    <dl>
+    <dt>'Mean[$list$]'
+    <dd>
+        Mean over all values in list (Total[list] / Length[list])
+    </dd>
+    </dt>
+    </dl>
+    >> Mean[{1,2,3}]
+     = 2
+    >> Mean[{a, b, 213}]
+     = (123 + a + b) / 3
+    """
+
+    rules = {
+        'Variance[list_List]': '(list-Mean[list]).Conjugate[list-Mean[list]]/(Length[list]-1)'
+    }
+
+
+class StandardDeviation(Builtin):
+    rules = {
+        'StandardDeviation[list_List]': 'Sqrt[Variance[list]]'
+    }
+
+class Quantile(Builtin):
+
+    rules = {
+        'Quantile[list_List, q_RealValueQ]': 'Sort[list, Less][[Ceiling[q * Length[list]]]]',
+        'Quantile[list_List, q_List]': 'Quantile[list, singleq] /@ q'
+    }
+
+
+class Diagonal(Builtin):
+    rules = {
+        'Diagonal[l_List, k]': 'Module[{k = 0, j = 0, i = 0}, If[n > 0, j = n, k = -n]; Table[l[[i + k, i + j]], {i, 1, Min[Dimensions[l] - {k, j}]}]]',
+        'Diagonal[l_List]' : 'Module[{i = 0}, Table[l[[i, i]], {i, 1, Min[Dimensions[l]]}]]'
+    }
+
+
+class Tr(Builtin):
```

Comment:
(assuming this is the TODO for Tr / Trace).

The Trace works in general, however, the Tr[m_List, head_, level_] doesn't, because Flatten is the wrong operator. So this needs to be fixed, or left out for now...

I think it needs to be something like partition, and then Fold the operator around the newly created list.